test/common/common.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2020 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving-operator/test/resources"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	mf "github.com/jcrossley3/manifestival"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// Mysteriously required to support GCP auth (required by k8s libs).
+	// Apparently just importing it is enough. @_@ side effects @_@.
+	// https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving-operator/test"
+)
+
+// Setup creates the client objects needed in the e2e tests.
+func Setup(t *testing.T) *test.Clients {
+	clients, err := test.NewClients(
+		pkgTest.Flags.Kubeconfig,
+		pkgTest.Flags.Cluster)
+	if err != nil {
+		t.Fatalf("Couldn't initialize clients: %v", err)
+	}
+	return clients
+}
+
+
+// KnativeServingVerify verifies if the KnativeServing can reach the READY status.
+func KnativeServingVerify(t *testing.T, clients *test.Clients, names test.ResourceNames) {
```

Comment:
im conflicted about the naming
i understand the CRD is called KnativeServing, but a method that says KnativeServingVerify is confusing for me.
maybe append CR at the end, as we are verifying the CR status.
thoughts?